### ISSUE TITLE
Horrendous Frame Lag

### ISSUE THREAD

**Issue by SweatyMeat** [SubnauticaNitrox/Nitrox#1107]
So I've recently setup a server for myself and a friend, port forwarding is fine we have no connection issues but there's a few bugs that we've encountered. One is that we get horrendous frame drops quite often. Sometimes not even when loading new areas, they can be at random. We also have issues where I'm appearing somewhere totally different to where I actually am on my friend's game. We get this thing where the seamoth randomly zips forward and blows up. Fish also seem to glitch out and not move.

Most of this doesn't bother me, but is there any way to fix the lag? My pc specs are pretty decent and I've hosted Minecraft servers in the past with no problems.

**Comment by flippo007** [comment]
The problem doesn't come from you, it's the server that still manages the **synchronization very badly** ;)

**Comment by SweatyMeat** [comment]
Are there any updates to fix sync problems, or is there anything I can do?

**Comment by Measurity** [comment]
Can you give measurements about how bad the performance is for you? How long are the lags? How many frames are dropping?

It would help if you can give us a reproducable scenario so that we can debug and find the performance issue.

**Comment by SweatyMeat** [comment]
So as I said before the lag spikes are very random so it's difficult to
tell you when exactly it will happen, but measurement wise I can help.

The lag spikes occur for about 2-5 seconds. Sometimes they can be really
quick, but most of them average around 3 seconds each. In terms of fps I go
down to 0 frames. The screen freezes entirely, then 3 seconds or so later
it goes back to normal.

**Comment by Sunrunner37** [comment]
Please try the latest version that was just released.  Previously, we had an overwhelming amount of messages that were being written to the player log file causing severe frame lag.  This could be the root cause of your issue.

There is still one type of frame lag around.  As you move around and reveal areas that you haven't previously seen the server has to load entities in them.  To mitigate this long term we will introduce a system to load all entities up front when the server starts but this is awhile off.

**Comment by SweatyMeat** [comment]
Is there a version that's been updated in the past week, because I setup this server around 4 days ago

**Comment by Sunrunner37** [comment]
We just released version 1.2.2.0.  You can check it out on our website or nexus.

**Comment by Jerakin** [comment]
Hi, I just played this with someone else and super impressed that it works so well that it does! The biggest issue we had were that the person connection via LAN had "lag spikes" that lasted 0.5 - 2 seconds, with a average frequency of around 10 seconds. It would happen on notifications (such as going down below 100m) and it also when opening their inventory. I connected with localhost and also had some lag spikes, not as frequent and not as long. I would notice them mainly when opening the inventory or chests.

Both our computers are good and can run single player mode without lag. We are both connected with a cable and I have forwarded the port (though it shouldn't matter as we are on LAN).
